docs/thrift.md

```diff
@@ -0,0 +1,205 @@
+Thrift over TChannel
+====================
+
+This document outlines how we intend to use Thrift over TChannel.
+
+For Thrift requests sent over TChannel, the `as` (arg scheme) transport header
+must be set to `thrift`. Requests will be made using `call req` messages and
+responses will be sent using `call res` messages, with values for `arg{1,2,3}`
+as defined in [Arguments][].
+
+For `call res`,
+
+-   In case of success, the Response Code (`code:1`) must be set to `0x00`.
+-   In case of failure, the Response Code (`code:1`) must be set to `0x01`.
+
+Arguments
```

Comment:
Should we add a note for what the `service:1` field should be ?

We should probably make it clear that it does not have to be thrift service but should be set (i.e. not an empty string).